rules_daml/daml.bzl

```diff
@@ -314,19 +314,41 @@ def daml_build_test(
 def daml_test(
         name,
         srcs = [],
+        deps = [],
         damlc = "//compiler/damlc",
         **kwargs):
     sh_inline_test(
         name = name,
-        data = [damlc] + srcs,
+        data = [damlc] + srcs + deps,
         cmd = """\
+set -eou pipefail
+tmpdir=$$(mktemp -d)
+trap "rm -rf $$tmpdir" EXIT
 DAMLC=$$(canonicalize_rlocation $(rootpath {damlc}))
 rlocations () {{ for i in $$@; do echo $$(canonicalize_rlocation $$i); done; }}
-
-$$DAMLC test --files $$(rlocations {files})
+DEPS=($$(rlocations {deps}))
+cat << EOF > $$tmpdir/daml.yaml
+sdk-version: {sdk_version}
+name: test
+version: 0.0.1
+source: .
+dependencies: [daml-stdlib, daml-prim $$([ $${{#DEPS[@]}} -gt 0 ] && printf ',"%s"' $${{DEPS[@]}})]
```

Comment:
I’m not quite clear why you need the `-gt 0` stuff. Shouldn’t this do the right thing for zero deps as well?

Reply:
For zero deps it prints `,""` which then fails with
```
damlc: user error (Cannot resolve SDK dependency ''. Use daml assistant.)
```